LandSandBoat/server
# [core, lua] Fix cure v and cura enmity

**_I affirm:_**

## What does this pull request do?
This PR fixes an issue whereby Cure V is giving twice the enmity it should (see https://github.com/LandSandBoat/server/issues/3016) and an issue whereby Cura spells are not being impacted by Tranquil Heart (because they were just using fixed enmity values from the SQL file). Retail testing shows that Tranquil Heart does impact Cure V so makes sense that it would also impact other fixed enmity cures like Cura.

This PR fixes these issues by passing optional fixed CE and VE values to the updateEnmityFromCure function (for cure V and cura-based spells) and then passing along these values until the final updateEnmityFromCure function in EnmityContainer which determines whether to use these fixed values or calculate dynamic values (based on cure total). The fixed enmity values in the SQL files are set to 0 like other cures (to prevent double counting).

Closes https://github.com/LandSandBoat/server/issues/3016

## Steps to test these changes
Attack a mob, then set your hp to 1, then cure yourself while monitoring the enmity of the mob towards you (for example via the !getenmity command). You should see for example that cure v is giving the correct enmity and cura is impacted by tranquil heart.
